modules/javafx.graphics/src/test/java/test/javafx/scene/text/TextTest.java

```diff
@@ -225,4 +232,41 @@
         assertNotNull(s);
         assertFalse(s.isEmpty());
     }
+
+    // Test for JDK-8130738
+    @Test public void testTabSize() {
+        Toolkit tk = (StubToolkit)Toolkit.getToolkit();
+        HBox root = new HBox();
+        Scene scene = new Scene(root);
+        Stage stage = new Stage();
+        stage.setScene(scene);
+        stage.setWidth(300);
+        stage.setHeight(200);
+
+        try {
+            Text text = new Text("\tHello");
+            root.getChildren().addAll(text);
+            stage.show();
+            tk.firePulse();
+            assertEquals(text.getTabSize(),8);
+            // initial width with default 8-space tab
+            double widthT8 = text.getBoundsInLocal().getWidth();
+            text.setTabSize(1);
+            tk.firePulse();
+            // width with tab at 1 spaces
+            double widthT1 = text.getBoundsInLocal().getWidth();
+            // approximate width of a single space
+            double widthSpace = (widthT8 - widthT1) / 7;
+            assertTrue(widthSpace > 0);
+            text.setTabSize(4);
+            tk.firePulse();
+            // width with tab at 4 spaces
+            double widthT4 = text.getBoundsInLocal().getWidth();
+            double expected = widthT8 - 4 * widthSpace;
+            // should be approximately 4 space-widths shorter
+            assertEquals(expected, widthT4, 0.5);
+        } finally {
+            stage.hide();
+        }
+  }
```

Comment:
In addition to the above test, I recommend some (very basic) tests of the setter / getting / property methods.